### ISSUE TITLE
[BUG] Select components don't handle integer values

### ISSUE THREAD

**Issue by gavinorland** [formio/formio.js#1755]
## Environment
Please provide as many details as you can:

- Hosting type
  - [x] Form.io
  - [ ] Local deployment
    - Version: 
- Formio.js version: v4.0.7 onward
- Frontend framework: React
- Browser: Chrome
- Browser version: 76.0.3809.132 (Official Build) (64-bit)

## Steps to Reproduce

1. Populate a Select component with objects that have integers instead of strings as their ID values. 
2. Interact with the menu a few times.

## Expected behavior

The menu should display the selected item's label - nothing else should happen.

## Observed behavior

Once an item has been selected, its value (the integer) is added as a new menu item. Furthermore, when the menu is opened again, this value is shown as the selected item's label.

## Example

The second menu is the one with integers.

http://jsfiddle.net/czguqLo6/1/

### Further info

This problem seems to have been introduced with v4.0.7 and we are seeing it manifested in various places in our application. For example, we also find that when a user adds a new item to a DataGrid, the Select component in the previous item of the grid will suddenly show its integer ID instead of its label.

**Comment by randallknutson** [comment]
We added a new option for this called "Storage Type" to the select component. Can you pick whether you want to store the value as a number or string?

We had some issues where customers were setting values like "01" which would be converted to an integer automatically and then stringified to compare the the values. "01" !== "1" so it would not find it to reset. By adding this we allow users to decide for themselves how they want to store the values instead of autotyping the values.

**Comment by aminmc** [comment]
@randallknutson if you set the data type to be number in the second select component you'll see this:

![Screenshot 2019-09-25 at 15 57 53](https://user-images.githubusercontent.com/290639/65613119-5c286c00-dfad-11e9-96a1-d67fb40e57b2.png)

Compared to the first one where it works as expected (setting the dataType to be string)

**Comment by gavinorland** [comment]
@randallknutson Thanks for getting back to me. Can you tell or show me where to find the "Storage Type" option? I'm not seeing it within our FormBuilder interface or in the underlying JSON.

**Comment by gavinorland** [comment]
Thanks for getting back to us so quickly, @randallknutson.

We've now tried setting "Storage Type" to string for the given Select component but unfortunately it has made no difference. I have an amended JSFiddle here where I've set a `dataType: 'string'` for the menu which uses integers so you can see this: http://jsfiddle.net/f9xkns8h/

Look forward to your response.

**Comment by aminmc** [comment]
@randallknutson Also jumping on this thread. Storage type at the component level doe not make a difference. 

<img width="1177" alt="Screenshot 2019-09-27 at 07 17 38" src="https://user-images.githubusercontent.com/290639/65746689-65255480-e0f7-11e9-8dce-d723470c0a21.png">

```json
{
  "key": "select",
  "data": {
    "url": "http://localhost:5000/v1/entities/grade",
    "headers": [
      {
        "key": "",
        "value": ""
      }
    ],
    "values": [],
    "json": "",
    "resource": "",
    "custom": ""
  },
  "type": "select",
  "input": true,
  "label": "Select",
  "widget": "choicesjs",
  "dataSrc": "url",
  "dataType": "number",
  "template": "<span>{{ item.test }}</span>",
  "validate": {
    "select": false,
    "required": false,
    "custom": "",
    "customPrivate": false
  },
  "indexeddb": {
    "table": "",
    "filter": "",
    "database": ""
  },
  "reference": false,
  "addResource": false,
  "searchField": "id__regex",
  "disableLimit": false,
  "selectValues": "data",
  "dataGridLabel": false,
  "valueProperty": "id",
  "selectThreshold": 0.3,
  "placeholder": "",
  "prefix": "",
  "customClass": "",
  "suffix": "",
  "multiple": false,
  "defaultValue": null,
  "protected": false,
  "unique": false,
  "persistent": true,
  "hidden": false,
  "clearOnHide": true,
  "refreshOn": "",
  "redrawOn": "",
  "tableView": true,
  "labelPosition": "top",
  "description": "",
  "errorLabel": "",
  "tooltip": "",
  "hideLabel": false,
  "tabindex": "",
  "disabled": false,
  "autofocus": false,
  "dbIndex": false,
  "customDefaultValue": "",
  "calculateValue": "",
  "attributes": {},
  "validateOn": "change",
  "conditional": {
    "show": null,
    "when": null,
    "eq": ""
  },
  "overlay": {
    "style": "",
    "left": "",
    "top": "",
    "width": "",
    "height": ""
  },
  "allowCalculateOverride": false,
  "encrypted": false,
  "alwaysEnabled": false,
  "showCharCount": false,
  "showWordCount": false,
  "properties": {},
  "allowMultipleMasks": false,
  "clearOnRefresh": false,
  "limit": 100,
  "lazyLoad": true,
  "filter": "",
  "searchEnabled": true,
  "minSearch": 0,
  "readOnlyValue": false,
  "authenticate": false,
  "selectFields": "",
  "searchThreshold": 0.3,
  "fuseOptions": {
    "include": "score",
    "threshold": 0.3
  },
  "customOptions": {},
  "id": "eqc121"
}
```

My json response:

{
  "schema": {

  },
  "data": [{
    "id":1,
    "test": "A"
  }, {
    "id": 2,
    "test": "B"
  },
    {
      "id": 3,
      "test": "C"
    },
    {
      "id": 4,
      "test": "D"
    },
    {
      "id": 5,
      "test": "E"
    }
  ]
}
```

**Comment by randallknutson** [comment]
This should fix the issue: https://github.com/formio/formio.js/pull/1783

**Comment by aminmc** [comment]
@randallknutson this is working as expected.

**Comment by gavinorland** [comment]
Thanks @randallknutson - look forward to next release.

**Comment by jhen1422** [comment]
This should be resolved in the latest 4.4.1 release.

**Comment by Ggleny** [comment]
Hi,  I have the same issue in react-formio 4.3.0.  if you  see in package.json you will find formio.js 5.XX  as a depdendecy https://github.com/formio/react/blob/master/package.json
